src/utils/uuid.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import crypto from 'crypto';
+
+/**
+ * Generates a random v4 UUID, as specified in RFC4122.
+ *
+ * Uses the native Web Crypto API if available, otherwise falls back to a
+ * polyfill.
+ *
+ * Example: '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d'
+ */
+export function uuidv4(): `${string}-${string}-${string}-${string}-${string}` {
+  // Available only in secure contexts
+  // https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API
+  if (crypto.randomUUID) {
```

Comment:
perhaps `if ('randomUUID' in globalThis.crypto)` and remove the import?

Reply:
Done